lib/process-release.js

```diff
@@ -0,0 +1,87 @@
+var semver = require('semver')
+var url = require('url')
+var path = require('path')
+
+var bitsre = /(x32|x64)/
+
+function processRelease (argv, gyp, defaultVersion, defaultRelease) {
+  var versionStr = argv[0] || gyp.opts.target || defaultVersion
+
+  // parse the version to normalize and ensure it's valid
+  var versionSemver = semver.parse(versionStr)
+  if (!versionSemver) {
+    return { versionStr: versionStr }
+  }
+  // flatten version into String
+  versionStr = versionSemver.version
+
+  var overrideDistUrl = gyp.opts['dist-url'] || gyp.opts.disturl
+  var iojs = !overrideDistUrl && !defaultRelease && semver.gte(versionStr, '1.0.0')
+  var name = (defaultRelease && defaultRelease.name) || (iojs ? 'iojs' : 'node')
+  var defaultDirUrl = (overrideDistUrl || iojs ? 'https://iojs.org/download/release' : 'https://nodejs.org/dist') + '/v' + versionStr + '/'
+  var baseUrl
+  var libUrl32
+  var libUrl64
+
+  if (defaultRelease && defaultRelease.headersUrl && !overrideDistUrl) {
+    baseUrl = url.resolve(defaultRelease.headersUrl, './')
+    libUrl32 = resolveLibUrl(name, defaultRelease.libUrl || baseUrl || defaultDirUrl, 32, versionStr)
+    libUrl64 = resolveLibUrl(name, defaultRelease.libUrl || baseUrl || defaultDirUrl, 64, versionStr)
+
+    return {
+      versionStr: versionStr,
+      versionSemver: versionSemver,
+      name: defaultRelease.name,
+      baseUrl: baseUrl,
+      tarballUrl: defaultRelease.headersUrl,
+      shasumsFile: 'SHASUMS256.txt',
+      shasumsUrl: url.resolve(baseUrl, 'SHASUMS256.txt'),
+      checksumAlgo: 'sha256',
+      versionDir: (defaultRelease.name != 'node' ? defaultRelease.name + '-' : '') + versionStr,
```

Comment:
done .. but you do know how silly that is right? also, do a grep for `==` vs `===` and `!=` vs `!==` in this codebase, there are no rules